File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/8 - 6 and 7 April 2019/06. Cinema Tickets.py
movie_name = None
free_spaces = 0
type_ticket = None
student = 0
standard = 0
kid = 0
total_sold = 0
current_move_total = 0

while True:
    movie_name = input()
    if movie_name == "Finish":
        break
    free_spaces = int(input())
    while True:
        if current_move_total == free_spaces:
            break
        type_ticket = input()
        if type_ticket == "End":
            break
        if type_ticket == "student":
            student += 1
            total_sold += 1
            current_move_total += 1
        elif type_ticket == "standard":
            standard += 1
            total_sold += 1
            current_move_total += 1
        elif type_ticket == "kid":
            kid += 1
            total_sold += 1
            current_move_total += 1
    print(f"{movie_name} - {(current_move_total / free_spaces * 100):.2f}% full.")
    current_move_total = 0

print(f"Total tickets: {total_sold}")
print(f"{(student / total_sold * 100):.2f}% student tickets.")
print(f"{(standard / total_sold * 100):.2f}% standard tickets.")
print(f"{(kid / total_sold * 100):.2f}% kids tickets.")

File: SoftUni Lessons/JavaScript Development/JS Basics May 2023/10. Nested Loops - Exercise/06. Cinema Tickets.js
function solution(params) {
    // MasK
    let tickets = { 'student': 0, 'standard': 0, 'kid': 0 };
    
    while('Finish') {
        let movie = params.shift();
        if (movie === 'Finish') { break; }
        let seats = Number(params.shift());
        let takenSeats = 0;
        while(seats !== takenSeats) {
            let ticket = params.shift();
            if (ticket === 'End') { break; }
            takenSeats += 1;
            tickets[ticket] += 1;
        }
        console.log(`${movie} - ${(takenSeats / seats * 100).toFixed(2)}% full.`);
    }

    let totalTickets = Object.values(tickets).reduce((a, b) => a + b, 0);
    
    console.log(`Total tickets: ${totalTickets}`);
    console.log(`${(tickets['student'] / totalTickets * 100).toFixed(2)}% student tickets.`);
    console.log(`${(tickets['standard'] / totalTickets * 100).toFixed(2)}% standard tickets.`);
    console.log(`${(tickets['kid'] / totalTickets * 100).toFixed(2)}% kids tickets.`);
}

solution(["Taxi",
"10",
"standard",
"kid",
"student",
"student",
"standard",
"standard",
"End",
"Scary Movie",
"6",
"student",
"student",
"student",
"student",
"student",
"student",
"Finish"]);

// Taxi - 60.00% full.
// Scary Movie - 100.00% full.
// Total tickets: 12
// 66.67% student tickets.
// 25.00% standard tickets.
// 8.33% kids tickets.

console.log('********************************');

solution(["The Matrix",
"20",
"student",
"standard",
"kid",
"kid",
"student",
"student",
"standard",
"student",
"End",
"The Green Mile",
"17",
"student",
"standard",
"standard",
"student",
"standard",
"student",
"End",
"Amadeus",
"3",
"standard",
"standard",
"standard",
"Finish"]);

// The Matrix - 40.00% full.
// The Green Mile - 35.29% full.
// Amadeus - 100.00% full.
// Total tickets: 17
// 41.18% student tickets.
// 47.06% standard tickets.
// 11.76% kids tickets.
